modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js

```diff
@@ -31,76 +32,102 @@ export default function Navigation({
 	timeSpanOptions,
 	viewURLs,
 }) {
-	const {endpoints, namespace, page, publishedToday, warning} = useContext(
-		StoreStateContext
-	);
+	const dispatch = useContext(StoreDispatchContext);
 
-	const {validAnalyticsConnection} = useContext(ConnectionContext);
+	const {
+		endpoints,
+		loading,
+		namespace,
+		page,
+		publishedToday,
+		trafficSources,
+		warning,
+	} = useContext(StoreStateContext);
 
-	const [currentPage, setCurrentPage] = useState({view: 'main'});
+	const {validAnalyticsConnection} = useContext(ConnectionContext);
 
-	const [trafficSources, setTrafficSources] = useState([]);
+	const {plid} = page;
 
-	const [trafficSourceName, setTrafficSourceName] = useState('');
+	const [currentPage, setCurrentPage] = useState({view: 'main'});
 
 	const chartState = useChartState();
 
 	const {timeSpanKey, timeSpanOffset} = chartState;
 
+	useEffect(() => {
+		const requests = Object.keys(endpoints).map((request) => {
+			if (request === 'analyticsReportsHistoricalReadsURL') {
+				return APIService.getHistoricalReads(
+					endpoints.analyticsReportsHistoricalReadsURL,
+					{
+						namespace,
+						plid,
+						timeSpanKey,
+						timeSpanOffset,
+					}
+				);
+			}
+			else if (request === 'analyticsReportsHistoricalViewsURL') {
+				return APIService.getHistoricalViews(
+					endpoints.analyticsReportsHistoricalViewsURL,
+					{
+						namespace,
+						plid,
+						timeSpanKey,
+						timeSpanOffset,
+					}
+				);
+			}
+			else if (request === 'analyticsReportsTotalViewsURL') {
+				return APIService.getTotalViews(
+					endpoints.analyticsReportsTotalViewsURL,
+					{namespace, plid}
+				);
+			}
+			else if (request === 'analyticsReportsTotalReadsURL') {
+				return APIService.getTotalReads(
+					endpoints.analyticsReportsTotalReadsURL,
+					{namespace, plid}
+				);
+			}
+			else if (request === 'analyticsReportsTrafficSourcesURL') {
+				return APIService.getTrafficSources(
+					endpoints.analyticsReportsTrafficSourcesURL,
+					{namespace, plid}
+				);
+			}
+		});
+
+		let metrics = {};
+
+		allSettled(requests)
+			.then((data) => {
+				for (var i = 0; i < data.length; i++) {
+					if (data[i].status === 'fulfilled') {
+						metrics = {
+							...metrics,
+							...data[i].value,
+						};
+					}
+					else {
+						dispatch({type: 'ADD_WARNING'});
```

Comment:
If more than one of your items wasn't fulfilled you would end up dispatching multiple actions here, which would probably be best consolidated into one. Also, it's not quite clear to me why you have two chained `then()` here. Maybe something like this?

```js
allSettled(requests).then((data) => {
    let addWarning = false;

    const metrics = data.reduce((result, {status, value}) => {
        if (status === 'fulfilled') {
            return {...result, ...value};
        }
        else {
            addWarning = true;
            return result;
        }
    }, {});

    if (addWarning) {
        dispatch({type: 'ADD_WARNING'});
    }

    dispatch({payload: metrics, type: 'SET_METRICS'});
});
```

Reply:
You're right, makes more sense to dispatch 'ADD_WARNING' only once although several endpoints failed. I'll change it ;) Thanks!